delphi-frontend/src/main/java/au/com/integradev/delphi/reporting/DelphiQuickFix.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Sonar Delphi Plugin
+ * Copyright (C) 2023 Integrated Application Development
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package au.com.integradev.delphi.reporting;
+
+import com.google.errorprone.annotations.FormatMethod;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.sonar.plugins.communitydelphi.api.ast.DelphiNode;
+import org.sonar.plugins.communitydelphi.api.token.DelphiToken;
+
+public class DelphiQuickFix {
+  private final String description;
+  private final List<DelphiTextEdit> textEdits;
+
+  private DelphiQuickFix(String description, List<DelphiTextEdit> textEdits) {
+    this.description = description;
+    this.textEdits = textEdits;
+  }
+
+  public String getDescription() {
+    return description;
+  }
+
+  public List<DelphiTextEdit> getTextEdits() {
+    return textEdits;
+  }
+
+  public static Builder newQuickFix(String description) {
+    return new Builder(description);
+  }
+
+  @FormatMethod
+  public static Builder newQuickFix(String description, Object... args) {
+    return new Builder(String.format(description, args));
+  }
+
+  public static class Builder {
+    private final String description;
+    private final List<DelphiTextEdit> textEdits = new ArrayList<>();
+
+    private Builder(String description) {
+      this.description = description;
+    }
+
+    public Builder addTextEdit(String replacement, DelphiNode node) {
```

Comment:
These are convenience routines to make the already clunky experience of writing a quick fix a little more ergonomic, since in the vast majority of cases you will want to be replacing a range exactly matching a node.

The alternative is that you have to specify a begin line, end line, begin column, and end column for every single text edit. For example, even this simple quick fix will become very unwieldy: 

```java
context
    .newIssue()
    .onNode(expression.getChild(0))
    .withMessage(MESSAGE)
    .withQuickFixes(
        DelphiQuickFix.newQuickFix("Remove redundant parentheses")
            .addTextEdit("", expression.getChild(0))
            .addTextEdit("", expression.getChild(2))
            .build())
    .report();
```

Unless you're suggesting we put these at another level? I'm not sure whatever level that would be, seeing as these are the builders used in checks.

Reply:
I'd suggest something like:
```java
context
    .newIssue()
    .onNode(expression.getChild(0))
    .withMessage(MESSAGE)
    .withQuickFixes(
        DelphiQuickFix.newQuickFix("Remove redundant parentheses")
            .addTextEdit(DelphiTextEdit.delete(expression.getChild(0)))
            .addTextEdit(DelphiTextEdit.delete(expression.getChild(2)))
            .build())
    .report();
```

Reply:
Hmm, it's a bit verbose, but if we're going to change the shape of the API a bit it could add some good flexibility.